src/types.ts

```diff
@@ -1886,6 +1910,10 @@ export class ZodObject<
     return deepPartialify(this) as any;
   }
 
+  deepStrict(): AnyZodObject {
```

Comment:
Sorry, this doesn't fly. The difficulty of typing the result of this method is why it doesn't exist. The schema hierarchy is always fully typed, all the way down, no exceptions. If people want a more "declarative" way to make their object schemas strict, you can use `z.strictObject` instead of `z.object`.

Reply:
Thanks for the feedback, two questions please. 

1st one is your opinion about approach on the same subject, proposed in https://github.com/colinhacks/zod/pull/1455
```typescript
schema.safeParse({
  deepStrict: true,
})
```
This could solve any issues with types and coverage, and allowed us to treat data as TypeScript does - without allowing optional keys.

2nd question - if you think that 1st case not an option - should I try to update all `strip` to `strict` in types of objects, returned by `deepStrict()`, or there are more deep concerns I'm not aware of?